flang/lib/Lower/CharacterExpr.cpp

```diff
@@ -471,7 +477,8 @@ bool Fortran::lower::CharacterExprHelper::isCharacter(mlir::Type type) {
   if (auto refType = type.dyn_cast<fir::ReferenceType>())
     type = refType.getEleTy();
   if (auto seqType = type.dyn_cast<fir::SequenceType>())
-    type = seqType.getEleTy();
+    if (seqType.getShape().size() == 1)
+      type = seqType.getEleTy();
```

Comment:
Why should an array of character not return true to `isCharacter` ?

Reply:
Because it isn't a scalar CHARACTER. Again, there appears to be no tests for arrays of CHARACTERs. There needs to be tests for the distinction rather than fuzzy logic.

Reply:
Arrays of CHARACTER can also be passed as `!fir.box`, which isn't checked here at all.

Reply:
Should we rename the function to `isCharacterScalar` then ?

Reply:
Please note, by "array of CHARACTER" I do not mean the type `!fir.array<?x!fir.char<?>>` which is the storage for a scalar CHARACTER variable.

Reply:
We can do that.